pool.go

```diff
@@ -3,22 +3,37 @@ package pool
 
 import (
 	"errors"
+	"time"
 )
 
 var (
 	// ErrClosed is the error resulting if the pool is closed via pool.Close().
-	ErrClosed = errors.New("pool is closed")
+	ErrClosed   = errors.New("pool is closed")
+	ErrTimedOut = errors.New("timed out waiting for connection")
 )
 
+type GenericConn interface{}
+
+type ConnectionHolder struct {
+	Conn  GenericConn
+	InUse bool
```

Comment:
Exactly  - the edge case is the same connection being returned to the pool twice. Probably hard to have happen now, but if I ever add timed expiration for a held connection this would come in handy.